File: packs/node-api/reference/src/routes/items.ts
import type { FastifyInstance } from 'fastify';
import { z } from 'zod';
import { listItems, getItem } from '../services/items.js';

const ItemParamsSchema = z.object({ id: z.string().min(1) });

export function itemRoutes(app: FastifyInstance): void {
  app.get('/items', async () => listItems());

  app.get('/items/:id', async (req, reply) => {
    const params = ItemParamsSchema.safeParse(req.params);
    if (!params.success) {
      return reply.status(400).send({ error: 'invalid id' });
    }
    const item = getItem(params.data.id);
    if (!item) {
      return reply.status(404).send({ error: 'not found' });
    }
    return item;
  });
}

File: packs/python-fastapi/reference/app/routers/items.py
from fastapi import APIRouter, HTTPException

from app.models import Item
from app.services.items import get_item, list_items

router = APIRouter()


@router.get("/items")
async def read_items() -> list[Item]:
    return list_items()


@router.get("/items/{item_id}")
async def read_item(item_id: int) -> Item:
    item = get_item(item_id)
    if item is None:
        raise HTTPException(status_code=404, detail="not found")
    return item
